core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -43,6 +43,7 @@ import org.apache.spark.rpc.RpcEnv
 import org.apache.spark.serializer.{Serializer, SerializerInstance}
 import org.apache.spark.shuffle.ShuffleManager
 import org.apache.spark.util._
+import org.apache.spark.util.collection.ReferenceCounter
```

Comment:
Can you add a comment in this class that explains the ref counting mechanism? It can be a shorter version of the commit message. 
Specifically:
  What are the invariants? (explain get()) Need to call release. What does it mean if it is 0?

I slightly prefer pin count over ref count (the block manager has a reference but it is unpinned)